src/amo/components/SearchForm/index.js

```diff
@@ -160,7 +166,7 @@ export class SearchFormBase extends React.Component {
       return;
     }
 
-    this.setState({ searchValue: '' }, () => {
+    this.setState({ autocompleteIsOpen: false, searchValue: '' }, () => {
       this.props.router.push(suggestion.url);
```

Comment:
I think you could safely remove `router.push()` from the `setState` callback too. I don't think it needs to wait for `setState` to resolve -- they are independent actions. It can be:

```js
this.setState({ autocompleteIsOpen: false, searchValue: '' });
this.props.router.push(suggestion.url);
```